c7n/schema.py

```diff
@@ -42,10 +43,12 @@
 def validate(data, schema=None):
     if schema is None:
         schema = generate()
-        Validator.check_schema(schema)
+
     validator = Validator(schema)
+    validator.check_schema(schema)
```

Comment:
schema validation should only be done at the time of generation, its big, not sure why this was changed.